### DISCUSSION TITLE
Is this web scrapping if you are accessing the json?

### DISCUSSION THREAD

**Discussion by D-Jeffrey** [crowbarz/aiopioneer#18] [Ideas]
Something of note.  [pioneer-receiver-notes](https://github.com/rwifall/pioneer-receiver-notes/blob/ec16e194f64860df4dcc2d105a57604a21a9c75c/README.md)  The `StatusHandlder.asp` give Zone and channel names.  Actually, it looks like a one stop current values page.  File this other interesting to know, but we have all that.

```
http://[192.168.4.44/StatusHandler.asp](http://192.168.4.44/StatusHandler.asp)

{
	"S":1,
	"B":0,
	"Z":[
		{"P":1,"V":93,"M":0,"I":[4,25,5,15,10,14,19,20,21,17,1,3,2,27,26,33],"C":5},
		{"P":0,"V":-1,"M":0,"I":[4,5,15,10,14,1,3,2,33],"C":4}
	],
	"L":1,
	"A":5,
	"IL":[
		"Wii           ",
		"Bluray        ",
		"TV/SAT",
		"DVR/BDR",
		"VIDEO 1",
		"VIDEO 2",
		"CabIe         ",
		"Computer      ",
		"Amazon FireTV ",
		"iPod/USB",
		"CD",
		"CD-R/TAPE",
		"TUNER",
		"SIRIUS",
		"INTERNET RADIO",
		"ADAPTER PORT"
	],
	"H":0
}
```
And the radio stations is this link for me. `InternetRadioSettingListHandler.asp`

```
 http://192.168.4.44/InternetRadioSettingListHandler.asp
{
	"R":[
	{
		"C":1,
		"T":"CBC Radio One",
		"U":"http://origin.www.cbc.ca/mrl2/livemedia/cbcr1-calgary.asx"
	},
	{
		"C":2,
		"T":"CFFR - 660 News",
		"U":"http://38.99.208.186/660NEWS"
	},
```
...
```
	{
		"C":24,
		"T":"FM Nirai",
		"U":"http://www.simulradio.jp/asx/fm-nirai.asx"
	}
	]
}	
```

**Comment by crowbarz** [comment]
A previous incarnation of the integration optionally used the HTTP interface to send commands to the AVR, however this was removed as it proved no more stable than the original Pioneer integration that opened a new TCP connection to the AVR every time it polled status or sent a command, as noted at the bottom of README.md. After a time, my AVR would completely stop accepting connections and a power cycle was needed to restore network connectivity. The occurrence of this issue was regularly within 48 hours of boot.

Using a single long-lived TCP connection to the AVR pseudo-serial interface resulted in far less instability. Also, the pseudo-serial interface streams state updates in real time, whereas the HTTP interface requires regular polling of the AVR to get updated state. That is also how the original Pioneer HA integration worked. Thus a design decision was made to use the pseudo-serial interface exclusively and abandon the HTTP interface.

**Comment by crowbarz** [comment]
[ADR0004](https://github.com/home-assistant/architecture/blob/master/adr/0004-webscraping.md) describes what Home Assistant considers to be webscraping and thus inadmissible into HA Core. Given the pages above return JSON, I believe it would be considered more as an HTTP based API rather than mimicking user interaction.

BTW the link above is already listed in the [aiopioneer README.md](https://github.com/crowbarz/aiopioneer#references)

**Comment by pantherale0** [comment]
I think the problem with older Pioneer AVRs is that they don't seem to clean down sessions too nicely server side once the client has (to a degree) "signed off", as already mentioned creating too many connections to the AVR will cause the networking stack to "crash" and to resolve requires a full power cycle of the AVR (unplug and plug in).

Web APIs would eventually create far too many web requests for the technology in the AVR to handle, I think aiopioneer should stay serial only as that is the most reliable and stable, and actually could prove to be the most compatible with older AVRs with only a serial interface and no LAN (you could create a bridge to expose the serial interface over telnet on a local network which this can then use to communicate with).

The SC-LX57 doesn't have a page at this address `http://avr/InternetRadioSettingListHandler.asp`, although the same thing may exist at a different location.

The StatusHandler.asp page doesn't appear to be much help at first glance, a list of source names, and perhaps some zone <-> source name mappings.
